API/app/Http/Controllers/ListPermissionsController.php

```diff
@@ -0,0 +1,126 @@
+<?php
+namespace App\Http\Controllers;
+
+use Illuminate\Routing\Controller as BaseController;
+
+use App\Models\SharedShoppingListPerm;
+use App\Models\SharedLink;
+use App\Models\ShoppingList;
+
+use Illuminate\Http\Request;  // Necessary to enforce Sanctum Middleware in constructor code
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\Auth;
+
+class ListPermissionsController extends BaseController
+{
+
+    // This constructor enforces the middleware for identifying a user via Laravel Sanctum
+    public function __construct()
+    {
+        $this->middleware('auth:sanctum');
+    }
+
+
+    //
+    public function share($id, Request $request)
+    {
+        $shoppingList = ShoppingList::findOrFail($id);
+        Log::info("Shopping list id: " . $shoppingList->list_id . " and we started with id: " . $id);
+        //Log::info("shoppingList object: " . print_r($shoppingList, true));
+
+        $maxRetries = 10; // Maximum number of retries
+        $retryCount = 0;
+
+        do {
+            $token = (string) Str::uuid();
+            $exists = SharedLink::where('token', $token)->exists();
+            $retryCount++;
+        } while ($exists && $retryCount < $maxRetries);
+
+        if ($exists) {
+            // If a unique token could not be generated, handle the error
+            return response()->json(['error' => 'Unable to generate a unique link. Please try again later.'], 500);
+        }
+
+        // Extract permissions from request
+        $canUpdate = $request->can_update ?? false;
+        $canDelete = $request->can_delete ?? false;
+
+        Log::info("canUpdate: " . $canUpdate);
+        Log::info("canDelete: " . $canDelete);
+
+
+        // Store the link with the shopping list ID and permissions
+        SharedLink::create([
+            'shopping_list_id' => $shoppingList->list_id,
+            'token' => $token,
+            'expires_at' => now()->addDays(7), // Optional: set expiration date
+            'can_update' => $canUpdate,
+            'can_delete' => $canDelete,
+        ]);
+
+        // Create the full URL with the domain
+        $url = "https://www.speedcartapp.com/share/$token";
+
+        return response()->json(['link' => $url]);
+    }
+
+    public function verifyShare($token, Request $request)
+    {
+        Log::info("Received shopping-lists request from IP address " . $request->ip());
+
+        // Grab user from sanctum
+        $user = Auth::user();
+
+        // Retrieve the shared link
+        $sharedLink = SharedLink::where('token', $token)->first();
+
+        // Check if the link exists
+        if (!$sharedLink) {
+            abort(404, 'Link not found');
+        }
+        Log::info("LINK WAS FOUND!");
```

Comment:
Change this log to only occur when `DEBUG_MODE` is on (create a constant for that in this file).